src/components/Dao/DaoSchemesPage.tsx

```diff
@@ -80,7 +80,7 @@ class DaoSchemesPage extends React.Component<IProps, null> {
     const allKnownSchemes = [...contributionReward, ...knownSchemes];
 
     const schemeManager = data[1];
-    const schemeManagerActive = getSchemeIsActive(schemeManager);
+    const schemeManagerActive = schemeManager ? getSchemeIsActive(schemeManager) : null;
```

Comment:
need to check scheme validity at getSchemeIsActive .
getSchemeIsActive should not crash alchemy in any case.

Reply:
I added comments for the `getSchemeIsActive` method and a check in `getSchemeIsActive` for `scheme` that throws a more clear exception when `scheme` is not set.

I can find no other incorrect uses of `getSchemeIsActive` in the app than the one I fixed.

Reply:
- when and how schemeManager can be null/false ? 
- how about getSchemeIsActive should return false if the scheme is null as null scheme is not active ?

Reply:
> * when and how schemeManager can be null/false ?

See the ticket this resolves: https://github.com/daostack/alchemy/issues/1888

> * how about getSchemeIsActive should return false if the scheme is null as null scheme is not active ?

That would be misleading, implying to the calling code that it has passed a valid scheme that is legitimately found to be not active, which is not the case.  Aberrant behavior would inevitably follow in the calling code.  The calling code must be responsible for properly handling the case that there is no scheme before calling this function.

Reply:
Throwing an exception on invalid parameters is a very common programming pattern.

Reply:
Throwing an exception which is manifested in the UI is a bad thing though

Reply:
Sure, but what you propose isn't the right way to address that.  There are already a jillions ways in which exceptions can occur because good code throws exceptions when unexpected things happen. We strive to program in a way that avoids them, but it is not feasible to eliminate all of them.  One of the things we do to handle the situation is to put up a relatively friendly error page, and in so doing, we are in effect handling all exceptions.